app/core/exceptions.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import logging
+from typing import NoReturn
+
+from fastapi import status as http_status
+
+from app.core.error_formats import create_error_response
+
+logger = logging.getLogger("api.exception")
+
+
+class UserCreationError(Exception):
+    """Custom exception for errors during user creation."""
+
+    pass
+
+
+class UserAccessTokenError(Exception):
+    """Custom exception for errors during user access token creation."""
+
+    pass
+
+
+class UserOrgCreationError(Exception):
+    """Custom exception for errors during organization creation."""
+
+    pass
+
+
+class OptScaleAPIResponseError(Exception):
+    """
+    Custom exception class for handling errors in the OptScale API responses.
+
+    Attributes:
+        status_code (int): The HTTP status code of the API response.
+        title (str): A short title describing the error.
+        reason (str): A detailed reason explaining the cause of the error.
+    """
+
+    def __init__(self, status_code: int, title: str, reason: str):
+        """
+        Initializes the exception with the given status code, title, and reason.
+
+        :param status_code: The HTTP status code of the API response.
+        :param title: A short title describing the error.
+        :param reason: A detailed reason explaining the cause of the error.
+        """
+        message = f"{title}"
+        super().__init__(message)
+        self.status_code = status_code
+        self.title = title
+        self.reason = reason
+
+
+def handle_exception(error: Exception) -> NoReturn:
+    """
+    Handles exceptions during user creation by logging the error and raising an
+    appropriate HTTP response.
+
+    :param error: error (Exception): The exception to handle.
+    Raises:
+        Exception: Raises a HTTPException exception with custom details of the exception.
+
+    Returns:
+        NoReturn: This function does not return; it always raises an exception.
+    """
+    logger.error(f"Exception occurred during user creation: {error}")
```

Comment:
It seems this is not a very well known behaviour of Python exceptions but while this looks intuitive, you'd end up with logs like this: `Exception occurred during user creation: None` (extreme but still realistic example) which as you can imagine it's not fun to debug 😆 

The reason for this is that when we cast an exception to a string, only the exception message is included and often that's not enough, e.g. "KeyError: None" is already a lot more useful with the exception class. And to make it even better, we can include the full stack trace which is surprisingly easy to do -- there is `exc_info ` keyword argument  on all logging methods which will do everything automatically by just setting it to `True` or the exception value: `logger.error("Exception occurred during user creation", exc_info=error)`. AND there is also `logger.exception` which is a simple wrapper for `logger.error(..., exc_info=True)`, I think it will work here too.